File: programming files/rgb_to_hex.java
public class rgb_to_hex {
    // public method takes in String[] args and returns nothing
    public static void main(String[] args) {
        // initialize r, g, and b
        int r = 255;
        int g = 127;
        int b = 0;
        
        // initialize hexColor as the returned value from rgbToHex()
        String hexColor = rgbToHex(r, g, b);
        
        // print out the values
        System.out.println("RGB color (" + r + ", " + g + ", " + b + ") = " + hexColor);
    }

    public static String rgbToHex(int r, int g, int b) {
        r = Math.min(255, Math.max(0, r));
        g = Math.min(255, Math.max(0, g));
        b = Math.min(255, Math.max(0, b));
        return String.format("%02X%02X%02X", r, g, b);
    }
}

//Test with RGB color (255, 127, 0) = FF7F00

File: programming files/rgb_to_hex.cc
//including the things we need
#include <iostream>
#include <sstream>
#include <iomanip>

std::string rgb_to_hex(int r, int g, int b)
{
    //assigning rgb values to r g and b
    r = std::max(0, std::min(255, r));
    g = std::max(0, std::min(255, g));
    b = std::max(0, std::min(255, b));

    //creating a string stream object
    std::stringstream ss;

    //adding color to string stream
    ss << std::uppercase << std::hex << std::setfill('0')
       << std::setw(2) << r << std::setw(2) << g << std::setw(2) << b;

    returign stringstream as a string
    return ss.str();
}

//Test with std::string hexColor = rgb_to_hex(255, 127, 0); // returns "FF7F00"
